Sources/Nats/NatsConnection.swift

```diff
@@ -511,8 +499,23 @@ class ConnectionHandler: ChannelInboundHandler {
     func close() async throws {
         self.reconnectTask?.cancel()
         await self.reconnectTask?.value
-        self.state = .closed
-        try await disconnect()
+
+        let eventLoop = self.channel?.eventLoop ?? MultiThreadedEventLoopGroup.currentEventLoop!
```

Comment:
should we guard these instead? What are the cases channel being nil other than `connect()` call not being done at this point?